library/constant_time_internal.h

```diff
@@ -46,7 +46,7 @@
  */
 unsigned mbedtls_ct_uint_mask( unsigned value );
 
-#if defined(MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC)
+#if defined(MBEDTLS_SSL_SOME_SUITES_USE_MAC)
```

Comment:
MAC is one definition _above_ CBC:
```
#if defined(MBEDTLS_SSL_SOME_SUITES_USE_STREAM) || \
    defined(MBEDTLS_SSL_SOME_SUITES_USE_CBC)
#define MBEDTLS_SSL_SOME_SUITES_USE_MAC
#endif
```
And because `mbedtls_ct_memcpy_if_eq` uses it (and is guarded by `MBEDTLS_SSL_SOME_SUITES_USE_MAC`) - I updated it. If something is guarded with `CBC` and uses it - it's OK, `MAC` (and therefore `mbedtls_ct_size_mask()`) will also be defined.
But looking at `ssl_msg.c` the whole block using `mbedtls_ct_size_mask_ge` -> `mbedtls_ct_size_mask_lt` -> `mbedtls_ct_size_mask` is guarded by `MBEDTLS_SSL_SOME_SUITES_USE_CBC`.

It just seems to me that due to:
```
#if defined(MBEDTLS_CIPHER_NULL_CIPHER)
#define MBEDTLS_SSL_SOME_SUITES_USE_STREAM
#endif
```
```
#if defined(MBEDTLS_SSL_SOME_SUITES_USE_STREAM) || \
    defined(MBEDTLS_SSL_SOME_SUITES_USE_CBC)
#define MBEDTLS_SSL_SOME_SUITES_USE_MAC
#endif
```
Message authentication can be done with a `NULL_CIPHER`, and in such case `mbedtls_ssl_decrypt_buf` still performs authentication, calling `mbedtls_ct_memcpy_offset` -> `mbedtls_ct_memcpy_if_eq` -> `mbedtls_ct_size_mask`.

So, to reiterate my understanding:
- `mbedtls_ct_size_mask` is used with `NULL_CIPHER` too, so `MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC` is not enough. Places that use `mbedtls_ct_size_mask` and are guarded with `MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC` are fine, since, if `MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC` is defined, `MBEDTLS_SSL_SOME_SUITES_USE_MAC` is defined too.

Reply:
Everything is now clear. Thanks.